### ISSUE TITLE
Unexpected behavior for matrix norm

### ISSUE THREAD

**Issue by jdebecdelievre** [pytorch/pytorch#49580]
Hello, 

I think that the matrix norm function does not behave as the API describes it. It computes a vector norm even when provided with a dimension.

```
import numpy as np
import torch

W = torch.tensor([[4.,0], [0 ,3.]])
print(f'Torch matrix norm: {W.norm(2, (0,1))}, Numpy matrix norm: {np.linalg.norm(W.numpy(), 2)}')
```

It returns the frobenius norm instead of the matrix 2-norm:
`Torch matrix norm: 5.0, Numpy matrix norm: 4.0`

Thanks!

  

cc @jlin27 @mruberry @jianyuh @nikitaved @pearu @heitorschueroff @walterddr @rgommers

**Comment by mruberry** [comment]
cc @kurtamohler 

I think the documentation is correct? When torch.norm is given a numeric norm it always computes a vector norm, I think, as indicated in the documentation here:

https://pytorch.org/docs/master/generated/torch.norm.html?highlight=norm#torch.norm

Note also that torch.norm is deprecated and torch.linalg.norm has behavior like NumPy's linalg.norm:

```
print(f'Torch matrix norm: {torch.linalg.norm(W, 2)}, Numpy matrix norm: {np.linalg.norm(W.numpy(), 2)}')
: Torch matrix norm: 4.0, Numpy matrix norm: 4.0
```

I think this is also consistent with the documentation for torch.linalg.norm:

https://pytorch.org/docs/master/linalg.html?highlight=linalg%20norm#torch.linalg.norm

**Comment by kurtamohler** [comment]
Yep, that's right @mruberry. The documentation describes this difference (at least from my perspective), but are there any specific details in the documentation that you think could be clarified or corrected, @jdebecdelievre ?

**Comment by jdebecdelievre** [comment]
It looks like the documentation for the current master branch is indeed correct, sorry about that.

I was using Pytorch 1.7.1,  whose documentation says that `torch.norm` will use matrix norm if `dim` is a 2-tuple (at least that is my understanding):
https://pytorch.org/docs/stable/generated/torch.norm.html
But I was getting the vector norm in my example below.

Thank you for your quick answer!

**Comment by mruberry** [comment]
Sorry for the confusion @jdebecdelievre. It's true that the documentation for torch.norm wasn't the best previously, but @kurtamohler really improved it.
